qiskit_experiments/calibration/experiments/rough_amplitude.py

```diff
@@ -0,0 +1,130 @@
+                 group: Optional[str] = 'default'):
+        """
+        Args:
+            qubit: The qubit on which to run the calibration.
+            cals: The instance to manage the calibrated pulse schedules and their
+                parameters.
+            gate_name: The name of the gate in cals to use when constructing the
+                rough amplitude calibration.
+            parameter_name: The name of the parameter to sweep, typically this will
+                be the amplitude of the pulse.
+            data_processor: The data processor used to process the measured data.
+            amplitudes: The amplitudes over which to sweep.
+            group: The calibration group that will be updated which defaults to 'default'.
+        """
+        circuit_options = ['initial_layout']
+        super().__init__([qubit], self.__class__.__name__, circuit_options)
+
+        self._cal_def = cals
+        self.amplitudes = amplitudes
+        self.parameter = parameter_name
+        self._qubit = qubit
+        self._data_processor = data_processor
+        self._calibration_group = group
+        self._gate_name = gate_name
+
+    def circuits(self, backend=None, **circuit_options) -> List[QuantumCircuit]:
+        """
+        Create the circuits for a rough rabi amplitude calibration and add the
+        required metadata.
+
+        Args:
+            backend (Backend): Not used.
+            circuit_options: Not used.
+
+        Returns:
+            A list of quantum circuits where a parameter is scanned.
+        """
+        circuits = []
+        for amplitude in self.amplitudes:
+            meta = CalibrationMetadata(
+                experiment_type=self._type,
+                pulse_schedule_name=self.__class__.__name__,
+                x_values=amplitude,
+                qubits=self._qubit
+            )
+
+            template_qc = self._cal_def.get_circuit(self._gate_name, (self._qubit,), [self.parameter])
+            template_qc.measure(0, 0)
+            template_qc.name = 'circuit'
+
+            qc = template_qc.assign_parameters({template_qc.parameters[0]: amplitude})
+            qc.metadata = asdict(meta)
+            circuits.append(qc)
+
+        return circuits
+
+    def update_calibrations(self, experiment_data: ExperimentData,
+                            update_pulses: Optional[Dict[str, float]] = None, index: int = -1):
+        """
+        Updates the amplitude of the pulses. This will preserve the existing phase.
+
+        Args:
+            experiment_data: The experiment data to use to update the pulse amplitudes.
+            update_pulses: The pulse to update. The key is the name of the pulse parameter and
+                the value is the angle to extract from the cosine fit. For example, {'amp_xp':
+                np.pi, 'amp_x90p': np.pi/2} will update the amplitude ot the xp and x90p pulses.
+            index: The index of analysis result to use in experiment_data. If this is not
```

Comment:
Because we need to extract the `experiment_id` from the `experiment_data`, see `exp_id=experiment_data.experiment_id` in the code. So either you give `(experiment_data, index)` or you give `(analysis_result, experiment_id)`.